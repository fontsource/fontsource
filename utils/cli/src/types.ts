export interface CLIOptions {
	test?: boolean;
	force?: boolean;
}

export interface BuildOptions {
	dir: string;
	tmpDir: string;
	force: boolean;
	isVariable: boolean;
}

export interface Axes {
	[axesType: string]: {
		default: string;
		min: string;
		max: string;
		step: string;
	};
}

export type VariableMetadata = boolean | Axes;

export type CategoryNames =
	| 'sans-serif'
	| 'serif'
	| 'display'
	| 'handwriting'
	| 'monospace'
	| 'icons'
	| 'other';

export type TypeNames = 'google' | 'league' | 'icons' | 'other';

export interface Metadata {
	id: string;
	family: string;
	subsets: string[];
	weights: number[];
	styles: string[];
	defSubset: string;
	variable: VariableMetadata;
	lastModified: string;
	version: string;
	category: CategoryNames;
	license: {
		type: string;
		url: string;
		attribution: string;
	};
	source: string;
	type: TypeNames;
}

export type UnicodeRange = {
	[subset: string]: string;
};
