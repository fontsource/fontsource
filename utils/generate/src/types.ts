export interface Axes {
	min: number | string;
	max: number | string;
}

export interface Source {
	url: string;
	format: string;
}

export interface Variable {
	wght?: Axes;
	stretch?: Axes; // wdth
	slnt?: Axes;
}

export interface FontObject {
	family: string;
	style: string;
	display: string;
	weight: number;
	src: Source[];
	variable?: Variable;
	unicodeRange?: string;
	comment?: string;
	spacer?: string;
}