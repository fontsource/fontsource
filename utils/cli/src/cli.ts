import 'dotenv/config';

import { cac } from 'cac';
import consola from 'consola';
import {
	fetchAPI,
	fetchVariable,
	generateAxis,
	parseLicenses,
	parsev1,
	parsev2,
	parseVariable } from 'google-font-metadata';
import colors from 'picocolors';

import { version } from '../package.json';
import { create } from './custom/create';
import { verify } from './custom/verify';
import { processGoogle } from './google/queue';

const cli = cac('fontsource');

cli
	.command('fetch [key]', 'Fetch parsing metadata for all fonts')
	.option('-f, --force', 'Force parse all metadata')
	.action(async (key: string, options) => {
		try {
			const finalKey = key ?? process.env.GOOGLE_API_KEY;
			if (!finalKey) {
				throw new Error('No API key provided.');
			}
			if (options.force) {
				consola.info(
					`Parsing all metadata... ${colors.bold(colors.red('[FORCE]'))}`
				);
			} else {
				consola.info('Parsing all metadata...');
			}
			await Promise.all([fetchAPI(finalKey), fetchVariable()]);
			await parsev1(options.force, false);
			await parsev2(options.force, false);
			await generateAxis();
			await parseVariable(false);
			await parseLicenses();
		} catch (error) {
			consola.error(error);
		}
	});

cli
	.command('build [...fonts]', 'Build font packages')
	.option('-v, --variable', 'Only build variable fonts')
	.option('-f, --force', 'Force rebuild all packages')
	.option('-t, --test', 'Build test fonts only')
	.action(async (fonts: string[], options) => {
		try {
			consola.info(`Building packages... ${options.force ? colors.bold(colors.red('[FORCE]')) : ''}`);
			await processGoogle(options, fonts);
		} catch (error) {
			consola.error(error);
		}
	});

cli.command('create').action(async () => {
	try {
		await create();
	} catch (error) {
		consola.error(error);
	}
});

cli.command('create-verify').action(async () => {
	try {
		await verify();
	} catch (error) {
		consola.error(error);
	}
});

cli.help();
cli.version(version);

cli.parse();
