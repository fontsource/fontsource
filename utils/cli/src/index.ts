export {
	APIIconStatic,
	APIIconVariable,
	APILicense,
	APIRegistry,
	APIv1,
	APIv2,
	APIVariable,
} from 'google-font-metadata';
