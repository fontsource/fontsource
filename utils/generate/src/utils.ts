// Find the lowest and highest value in a weight array
export const findBoundaries = (weights: number[]) => [Math.min(...weights), Math.max(...weights)];
