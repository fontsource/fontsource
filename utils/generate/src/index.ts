import type { FontMetadata, FontObject } from './types';
import { findBoundaries } from './utils';

const generateSingle = (font: FontObject) => {
	const { family, style, display, weight, variable, src, unicodeRange, comment, spacer = '\n  ' } = font;
	// If variable, modify output
	const { wght, stretch, slnt } = variable ?? {};
	let result = '@font-face {';
	result += `${spacer}font-family: '${family}';`;

	// If slnt is present, switch to oblique style
	result += `${spacer}font-style: ${slnt ? `oblique ${slnt.max * -1}deg ${slnt.min * -1}deg` : style};`;
	result += `${spacer}font-display: ${display};`;
	// If variable wght is present, use min max wght vals
	result += `${spacer}font-weight: ${wght ? findBoundaries(wght).join(' ') : weight};`;

	if (stretch)
		result += `${spacer}font-stretch: ${stretch.min}% ${stretch.max}%;`;

	// Merges all formats into a single src
	result += `${spacer}src: ${src.map(({ url, format }) => `url(${url}) format('${format}')`).join(', ')};`;

	if (unicodeRange)
		result += `${spacer}unicode-range: ${unicodeRange};`;

	if (comment)
		result = `/* ${comment} */\n${result}`;

	return `${result}\n}`;
};

const generateMulti = (metadata: FontMetadata) => {
	const { family, styles, display, weights, variable, path = '', subsets, unicodeRange, formats } = metadata;
	const id = metadata.id ?? family.toLowerCase().replace(/\s/g, '-');
	let result = '';
	for (const style of styles) {
		for (const weight of weights) {
			for (const subset of subsets) {
				const fontObject: FontObject = {
					family,
					style,
					display,
					weight,
					variable,
					src: formats.map(format => ({ url: `${path}${id}-${subset}-${weight}-${style}.${format}`, format })),
					comment: subset
				};

				if (unicodeRange)
					fontObject.unicodeRange = unicodeRange[subset];

				result += `${generateSingle(fontObject)}\n\n`;
			}
		}
	}
	// Slice last 2 \n if no minify
	return result.slice(0, -2);
};

export { generateMulti, generateSingle };