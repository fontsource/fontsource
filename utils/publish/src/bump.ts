import consola from 'consola';
import colors from 'picocolors';
import prompts from 'prompts';
import semver from 'semver';

import type { BumpFlags, BumpObject } from './types';
import { bumpCheck, findDiff, mergeFlags, pathToPackage } from './utils';

const isValidBumpArg = (bumpArg: string): boolean => {
	const validBumpArgs = new Set(['patch', 'minor', 'major', 'from-package']);

	// If it isn't a bumpArg in the set and isn't a semver version number
	if (!validBumpArgs.has(bumpArg) && !semver.valid(bumpArg)) {
		return false;
	}
	return true;
};

const bumpValue = (oldVersion: string, bumpArg: string): string | false => {
	// Check if valid semver version and if invalid return false
	const arr = semver.valid(oldVersion)?.split('.');
	if (arr) {
		if (bumpArg === 'patch') {
			const newNum = Number(arr[2]) + 1;
			arr[2] = String(newNum);
			return arr.join('.');
		}
		if (bumpArg === 'minor') {
			const newNum = Number(arr[1]) + 1;
			arr[1] = String(newNum);
			return arr.join('.');
		}
		if (bumpArg === 'major') {
			const newNum = Number(arr[0]) + 1;
			arr[0] = String(newNum);
			return arr.join('.');
		}
		if (bumpArg === 'from-package') {
			return arr.join('.');
		}
		// Else just return number version
		if (semver.valid(bumpArg)) {
			return bumpArg;
		}

		return false;
	}
	return false;
};

const createBumpObject = async (diff: string[], bumpArg: string): BumpObject[] => {
	const bumpObjectArr: BumpObject[] = [];
	const packageJsons = await pathToPackage(diff);

	let index = 0;
	for (const filePath of diff) {
		const packageJson = packageJsons[index];
		const bumpedVersion = bumpValue(packageJson.version, bumpArg);

		// If bumped version returns false, set failedValidation to true
		const bumpObject: BumpObject = bumpedVersion
			? {
				packageFile: packageJson,
				packagePath: filePath,
				bumpedVersion,
			}
			: {
				packageFile: packageJson,
				packagePath: filePath,
				bumpedVersion,
				noPublish: true,
			};

		bumpObjectArr.push(bumpObject);
		index += 1;
	}

	return bumpObjectArr;
};

const bumpVerify = async (
	bumpFlagVar: FlagsBumpReturn,
	bumpObjects: BumpObject[],
	bumpArg: string
): Promise<BumpObject[]> => {
	// Skip checking if no verify flag
	let checkedObjects: BumpObject[];
	if (bumpFlagVar.noVerify) {
		consola.info(colors.red('Skipping version verification due to noVerify flag...'));
		checkedObjects = bumpObjects;
	} else {
		checkedObjects = await bumpCheck(bumpObjects, bumpArg);
	}

	if (checkedObjects.length === 0) {
		throw new Error(colors.bold(colors.red('No packages to update found.')));
	}

	consola.info(colors.bold(colors.blue(('Changed packages:'))));
	for (const bumpObject of checkedObjects) {
		consola.info(
			colors.magenta(
				`${bumpObject.packageFile.name}: ${bumpObject.packageFile.version} --> ${bumpObject.bumpedVersion}`
			)
		);
	}

	if (!bumpFlagVar.skipPrompt) {
		// Filter out any objects with the noPublish flag attached to them
		checkedObjects = checkedObjects.filter(
			bumpObject => !bumpObject.noPublish
		);
		const input = await prompts({
			type: 'confirm',
			name: 'value',
			message: 'Bump packages?',
			initial: true
		});
		if (!input.value) {
			throw new Error(colors.bold(colors.red('Bump cancelled.')));
		}
	}

	return checkedObjects;
};

const bump = async (version: string, options: BumpFlags) => {
	if (!isValidBumpArg(version)) {
		throw new Error('Incorrect bump argument.');
	}
	consola.info(`${colors.bold(colors.blue('Bumping packages...'))} ${options.forcePublish ? colors.red(colors.bold('[FORCE]')) : ''}`);
	const config = await mergeFlags(options);
	const diff = await findDiff(config);
	const bumpObjects = await createBumpObject(diff, version);
	const checkedObjects = await bumpCliPrint(
		bumpFlagVars,
		bumpObjects,
		bumpArg
	);
};

export { bump };
