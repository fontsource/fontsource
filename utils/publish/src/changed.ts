import consola from 'consola';
import colors from 'picocolors';

import type { ChangedFlags } from './types';
import { findDiff, mergeFlags, pathToPackage } from './utils';

const changed = async (options: ChangedFlags) => {
	consola.info(colors.bold(colors.blue('Checking packages...')));
	// CLI flags take precedence over default config
	const config = await mergeFlags(options);
	const diff = await findDiff(config);
	const packages = await pathToPackage(diff);

	if (packages.length > 0) {
		consola.info(colors.bold(colors.blue('Packages changed:')));
		for (const packageJson of packages) {
			consola.info(colors.bold(`${packageJson}`));
		}
	} else {
		consola.info(colors.bold(colors.red('No packages changed.')));
	}
};

export { changed };
