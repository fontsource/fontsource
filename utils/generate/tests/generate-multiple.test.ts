import { describe, expect, it } from 'vitest';

import { generateMulti } from '../src';


describe('generateMulti', () => {
	it('should generate a multi font-face', () => {
		const font = {
			family: 'Open Sans',
			id: 'open-sans',
			styles: ['normal', 'italic'],
			display: 'swap',
			weights: [400, 700],
			subsets: ['latin', 'latin-ext'],
			formats: ['woff2', 'woff', 'ttf']
		};

		const result = generateMulti(font);
		expect(result).toMatchSnapshot();
	});

	it('should generate a multi font-face with different display', () => {
		const font = {
			family: 'Open Sans',
			id: 'open-sans',
			styles: ['normal', 'italic'],
			display: 'optional',
			weights: [400, 700],
			subsets: ['latin', 'latin-ext'],
			formats: ['woff2', 'woff', 'ttf']
		};

		const result = generateMulti(font);
		expect(result).toMatchSnapshot();
	});

	it('should generate a multi font-face with additional path', () => {
		const font = {
			family: 'Open Sans',
			id: 'open-sans',
			styles: ['normal', 'italic'],
			display: 'swap',
			weights: [400, 700],
			subsets: ['latin', 'latin-ext'],
			formats: ['woff2', 'woff', 'ttf'],
			path: './fonts/'
		};

		const result = generateMulti(font);
		expect(result).toMatchSnapshot();
	});

	it('should generate with unicode range', () => {
		const font = {
			family: 'Open Sans',
			id: 'open-sans',
			styles: ['normal', 'italic'],
			display: 'swap',
			weights: [400, 700],
			subsets: ['latin', 'latin-ext'],
			formats: ['woff2', 'woff', 'ttf'],
			unicodeRange: {
				'latin': 'U+000-5FF',
				'latin-ext': 'U+000-5FF2'
			}
		};

		const result = generateMulti(font);
		expect(result).toMatchSnapshot();
	});

	it('should generate a multi font-face with all variable', () => {
		const font = {
			family: 'Open Sans',
			id: 'open-sans',
			styles: ['normal'],
			display: 'swap',
			weights: [100, 700, 900],
			subsets: ['latin', 'latin-ext'],
			formats: ['woff2', 'woff', 'ttf'],
			variable: {
				wght: [100, 700, 900],
				stretch: {
					min: 50,
					max: 200
				},
				slnt: {
					min: -10,
					max: 20
				}
			}
		};

		const result = generateMulti(font);
		expect(result).toMatchSnapshot();
	});
});
