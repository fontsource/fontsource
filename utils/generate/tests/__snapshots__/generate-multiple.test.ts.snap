// Vitest Snapshot v1

exports[`generateMulti > should generate a multi font-face 1`] = `
"/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-400-normal.woff2) format('woff2'), url(open-sans-latin-400-normal.woff) format('woff'), url(open-sans-latin-400-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-ext-400-normal.woff2) format('woff2'), url(open-sans-latin-ext-400-normal.woff) format('woff'), url(open-sans-latin-ext-400-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-700-normal.woff2) format('woff2'), url(open-sans-latin-700-normal.woff) format('woff'), url(open-sans-latin-700-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-ext-700-normal.woff2) format('woff2'), url(open-sans-latin-ext-700-normal.woff) format('woff'), url(open-sans-latin-ext-700-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-400-italic.woff2) format('woff2'), url(open-sans-latin-400-italic.woff) format('woff'), url(open-sans-latin-400-italic.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-ext-400-italic.woff2) format('woff2'), url(open-sans-latin-ext-400-italic.woff) format('woff'), url(open-sans-latin-ext-400-italic.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-700-italic.woff2) format('woff2'), url(open-sans-latin-700-italic.woff) format('woff'), url(open-sans-latin-700-italic.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-ext-700-italic.woff2) format('woff2'), url(open-sans-latin-ext-700-italic.woff) format('woff'), url(open-sans-latin-ext-700-italic.ttf) format('ttf');
}"
`;

exports[`generateMulti > should generate a multi font-face with additional path 1`] = `
"/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./fonts/open-sans-latin-400-normal.woff2) format('woff2'), url(./fonts/open-sans-latin-400-normal.woff) format('woff'), url(./fonts/open-sans-latin-400-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./fonts/open-sans-latin-ext-400-normal.woff2) format('woff2'), url(./fonts/open-sans-latin-ext-400-normal.woff) format('woff'), url(./fonts/open-sans-latin-ext-400-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url(./fonts/open-sans-latin-700-normal.woff2) format('woff2'), url(./fonts/open-sans-latin-700-normal.woff) format('woff'), url(./fonts/open-sans-latin-700-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url(./fonts/open-sans-latin-ext-700-normal.woff2) format('woff2'), url(./fonts/open-sans-latin-ext-700-normal.woff) format('woff'), url(./fonts/open-sans-latin-ext-700-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./fonts/open-sans-latin-400-italic.woff2) format('woff2'), url(./fonts/open-sans-latin-400-italic.woff) format('woff'), url(./fonts/open-sans-latin-400-italic.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./fonts/open-sans-latin-ext-400-italic.woff2) format('woff2'), url(./fonts/open-sans-latin-ext-400-italic.woff) format('woff'), url(./fonts/open-sans-latin-ext-400-italic.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 700;
  src: url(./fonts/open-sans-latin-700-italic.woff2) format('woff2'), url(./fonts/open-sans-latin-700-italic.woff) format('woff'), url(./fonts/open-sans-latin-700-italic.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 700;
  src: url(./fonts/open-sans-latin-ext-700-italic.woff2) format('woff2'), url(./fonts/open-sans-latin-ext-700-italic.woff) format('woff'), url(./fonts/open-sans-latin-ext-700-italic.ttf) format('ttf');
}"
`;

exports[`generateMulti > should generate a multi font-face with all variable 1`] = `
"/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: oblique -20deg 10deg;
  font-display: swap;
  font-weight: 100 900;
  font-stretch: 50% 200%;
  src: url(open-sans-latin-100-normal.woff2) format('woff2'), url(open-sans-latin-100-normal.woff) format('woff'), url(open-sans-latin-100-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: oblique -20deg 10deg;
  font-display: swap;
  font-weight: 100 900;
  font-stretch: 50% 200%;
  src: url(open-sans-latin-ext-100-normal.woff2) format('woff2'), url(open-sans-latin-ext-100-normal.woff) format('woff'), url(open-sans-latin-ext-100-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: oblique -20deg 10deg;
  font-display: swap;
  font-weight: 100 900;
  font-stretch: 50% 200%;
  src: url(open-sans-latin-700-normal.woff2) format('woff2'), url(open-sans-latin-700-normal.woff) format('woff'), url(open-sans-latin-700-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: oblique -20deg 10deg;
  font-display: swap;
  font-weight: 100 900;
  font-stretch: 50% 200%;
  src: url(open-sans-latin-ext-700-normal.woff2) format('woff2'), url(open-sans-latin-ext-700-normal.woff) format('woff'), url(open-sans-latin-ext-700-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: oblique -20deg 10deg;
  font-display: swap;
  font-weight: 100 900;
  font-stretch: 50% 200%;
  src: url(open-sans-latin-900-normal.woff2) format('woff2'), url(open-sans-latin-900-normal.woff) format('woff'), url(open-sans-latin-900-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: oblique -20deg 10deg;
  font-display: swap;
  font-weight: 100 900;
  font-stretch: 50% 200%;
  src: url(open-sans-latin-ext-900-normal.woff2) format('woff2'), url(open-sans-latin-ext-900-normal.woff) format('woff'), url(open-sans-latin-ext-900-normal.ttf) format('ttf');
}"
`;

exports[`generateMulti > should generate a multi font-face with different display 1`] = `
"/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: optional;
  font-weight: 400;
  src: url(open-sans-latin-400-normal.woff2) format('woff2'), url(open-sans-latin-400-normal.woff) format('woff'), url(open-sans-latin-400-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: optional;
  font-weight: 400;
  src: url(open-sans-latin-ext-400-normal.woff2) format('woff2'), url(open-sans-latin-ext-400-normal.woff) format('woff'), url(open-sans-latin-ext-400-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: optional;
  font-weight: 700;
  src: url(open-sans-latin-700-normal.woff2) format('woff2'), url(open-sans-latin-700-normal.woff) format('woff'), url(open-sans-latin-700-normal.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: optional;
  font-weight: 700;
  src: url(open-sans-latin-ext-700-normal.woff2) format('woff2'), url(open-sans-latin-ext-700-normal.woff) format('woff'), url(open-sans-latin-ext-700-normal.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: optional;
  font-weight: 400;
  src: url(open-sans-latin-400-italic.woff2) format('woff2'), url(open-sans-latin-400-italic.woff) format('woff'), url(open-sans-latin-400-italic.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: optional;
  font-weight: 400;
  src: url(open-sans-latin-ext-400-italic.woff2) format('woff2'), url(open-sans-latin-ext-400-italic.woff) format('woff'), url(open-sans-latin-ext-400-italic.ttf) format('ttf');
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: optional;
  font-weight: 700;
  src: url(open-sans-latin-700-italic.woff2) format('woff2'), url(open-sans-latin-700-italic.woff) format('woff'), url(open-sans-latin-700-italic.ttf) format('ttf');
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: optional;
  font-weight: 700;
  src: url(open-sans-latin-ext-700-italic.woff2) format('woff2'), url(open-sans-latin-ext-700-italic.woff) format('woff'), url(open-sans-latin-ext-700-italic.ttf) format('ttf');
}"
`;

exports[`generateMulti > should generate with unicode range 1`] = `
"/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-400-normal.woff2) format('woff2'), url(open-sans-latin-400-normal.woff) format('woff'), url(open-sans-latin-400-normal.ttf) format('ttf');
  unicode-range: U+000-5FF;
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-ext-400-normal.woff2) format('woff2'), url(open-sans-latin-ext-400-normal.woff) format('woff'), url(open-sans-latin-ext-400-normal.ttf) format('ttf');
  unicode-range: U+000-5FF2;
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-700-normal.woff2) format('woff2'), url(open-sans-latin-700-normal.woff) format('woff'), url(open-sans-latin-700-normal.ttf) format('ttf');
  unicode-range: U+000-5FF;
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-ext-700-normal.woff2) format('woff2'), url(open-sans-latin-ext-700-normal.woff) format('woff'), url(open-sans-latin-ext-700-normal.ttf) format('ttf');
  unicode-range: U+000-5FF2;
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-400-italic.woff2) format('woff2'), url(open-sans-latin-400-italic.woff) format('woff'), url(open-sans-latin-400-italic.ttf) format('ttf');
  unicode-range: U+000-5FF;
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(open-sans-latin-ext-400-italic.woff2) format('woff2'), url(open-sans-latin-ext-400-italic.woff) format('woff'), url(open-sans-latin-ext-400-italic.ttf) format('ttf');
  unicode-range: U+000-5FF2;
}

/* latin */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-700-italic.woff2) format('woff2'), url(open-sans-latin-700-italic.woff) format('woff'), url(open-sans-latin-700-italic.ttf) format('ttf');
  unicode-range: U+000-5FF;
}

/* latin-ext */
@font-face {
  font-family: 'Open Sans';
  font-style: italic;
  font-display: swap;
  font-weight: 700;
  src: url(open-sans-latin-ext-700-italic.woff2) format('woff2'), url(open-sans-latin-ext-700-italic.woff) format('woff'), url(open-sans-latin-ext-700-italic.ttf) format('ttf');
  unicode-range: U+000-5FF2;
}"
`;
